libcaf_core/caf/flow/observable.hpp

```diff
@@ -269,6 +270,12 @@ class observable_def {
     return materialize().buffer(count, period);
   }
 
+  /// @copydoc observable::catch
+  template <class Predicate, class Fallback>
+  auto catch_and_resume(Predicate predicate, Fallback fallback) {
+    return materialize().catch_and_resume(predicate, std::move(fallback));
```

Comment:
```suggestion
  template <class Predicate, class Fallback>
  auto catch_and_resume(Predicate&& predicate, Fallback&& fallback) {
    return materialize().catch_and_resume(std::forward<Predicate>(predicate),
                                          std::forward<Fallback>(fallback));
```

Copying lambdas will not work when the lambda captures a non-copyable type, e.g., because it captured a unique pointer.

This also applies to other places throughout the code in this PR, and also many other flow operators in general (which I hope can be fixed for CAF 2.x).

See also:
- https://github.com/actor-framework/actor-framework/issues/1968